src/gui/shutdownconfirm.cpp

```diff
@@ -33,50 +33,81 @@
 #include "base/types.h"
 #include "shutdownconfirm.h"
 
+#include <QVBoxLayout>
+#include <QHBoxLayout>
+#include <QStyle>
+#include <QIcon>
+#include <QLabel>
+#include <QDialogButtonBox>
+#include <QCheckBox>
 #include <QPushButton>
 
 ShutdownConfirmDlg::ShutdownConfirmDlg(const ShutdownAction &action)
-    : m_exitNow(0)
+    : m_neverShowAgain(false)
     , m_timeout(15)
     , m_action(action)
 {
-    // Title and button
+    QVBoxLayout *myLayout = new QVBoxLayout();
+    //Warning Icon and Text
+    QHBoxLayout *messageRow = new QHBoxLayout();
+    QLabel *warningLabel = new QLabel();
+    QIcon warningIcon(style()->standardIcon(QStyle::SP_MessageBoxWarning, 0, this));
+    warningLabel->setPixmap(warningIcon.pixmap(warningIcon.actualSize(QSize(32, 32))));
+    messageRow->addWidget(warningLabel);
+    m_text = new QLabel();
+    messageRow->addWidget(m_text);
+    myLayout->addLayout(messageRow);
+    updateText();
+    QDialogButtonBox *buttons = new QDialogButtonBox(Qt::Horizontal);
+    // Never show again checkbox, Title, and button
     if (m_action == ShutdownAction::None) {
+        //Never show again checkbox (shown only when exitting without shutdown)
+        QCheckBox *neverShowAgainCheckbox = new QCheckBox(tr("Never show again"));
+        myLayout->addWidget(neverShowAgainCheckbox, 0, Qt::AlignHCenter);
+        //Title and button
+        connect(neverShowAgainCheckbox, SIGNAL(clicked(bool)), this, SLOT(handleNeverShowAgainCheckboxToggled(bool)));
```

Comment:
You don't have to connect to the signal/slot here. When the dialog is closed, before returning succes/failure -> also check the state of `neverShowAgainCheckBox`